# Single-event probability problems

Assum 100,000 simulations ran for each question.

In [1]:
# set my imprts
import numpy as np

# set my random seed
np.random.seed(123)

# set global iterations
iterations = 100_000

1. If you roll a fair six-sided die, what is the probability of rolling a 3?

In [2]:
# numbers contained by my die
die = np.arange(1,7)

# run iterations and make random choices
rand_selection = np.random.choice(a= die, size=iterations)

# count all 3 in the rand_selection
rand_selection = (rand_selection == 3).sum()

# get the mean
rand_selection.mean() / iterations

0.17018

    - The probability of rolling a 3 on a fair six-sided die is 1/6 or approximately 0.1657.

2. A jar contains 10 red balls and 5 blue balls. If you randomly select one ball from the jar, what is the probability of selecting a red ball?.

In [3]:
# selection options in the jar
jar = ["red", "blue"]

# assign weights to each option in the jar
weights = [10/15, 5/15]

# run iterations and make random choices based on given weights
rand_selection = np.random.choice(a=jar,size=iterations,p=weights)

# count all red balls in the jar
rand_selection = (rand_selection == "red").sum()

# get the mean and persantage
rand_selection.mean() / iterations

0.66586

    - The probability of selecting a red ball from a jar containing 10 red balls and 5 blue balls is 10/15 or approximately 0.668.
    
3. Two cards are drawn successively (without replacement) from a standard deck of 52 playing cards. What is the probability that the first card drawn is a heart and the second card drawn is a diamond?

In [4]:
# set card draw options
card_draw_1 = ["heart", "Other"]
weights_1 = [13/52, 39/52]

card_draw_2 = ["diamond", "Other"]
weights_2 = [13/51, 38/51] # 51 beacuse card 1 was not replaced

# run iterations and make random choices based on given weights
rand_selection_1 = np.random.choice(a= card_draw_1, size=iterations, p=weights_1)
rand_selection_2 = np.random.choice(a= card_draw_2, size=iterations, p=weights_2)

# count all hearts and diamonds
rand_selection_1 = (rand_selection_1 == "heart").sum()
rand_selection_2 = (rand_selection_2 == "diamond").sum()

# get the mean and persantage
rand_selection_1 = rand_selection_1.mean() / iterations
rand_selection_2 = rand_selection_2.mean() / iterations

# multiply the probalilities
rand_selection_1 * rand_selection_2

0.062476609

    - The probability of drawing a heart followed by a diamond from a standard deck of 52 playing cards, without replacement, is (13/52) * (13/51) or approximately 0.0638.

4. A box contains 4 red marbles, 3 blue marbles, and 2 green marbles. If you randomly select two marbles from the box without replacement, what is the probability of selecting a red marble first and then a blue marble?

In [5]:
# set marble draw events
marble_draw_1 = ["red", "Other"]
weights_1 = [4/9, 5/9]

marble_draw_2 = ["blue", "Other"]
weights_2 = [3/8, 5/8] # 8 beacuse marble 1 was not replaced

# run iterations and make random choices based on given weights
rand_selection_1 = np.random.choice(a= marble_draw_1, size=iterations, p=weights_1)
rand_selection_2 = np.random.choice(a= marble_draw_2, size=iterations, p=weights_2)

# count all red and blue marbles
rand_selection_1 = (rand_selection_1 == "red").sum()
rand_selection_2 = (rand_selection_2 == "blue").sum()

# get probability
rand_selection_1 = rand_selection_1.mean() / iterations
rand_selection_2 = rand_selection_2.mean() / iterations

# multiply the probalilities
rand_selection_1 * rand_selection_2

0.16557418429999998

    - The probability of selecting a red marble followed by a blue marble from a box containing 4 red marbles, 3 blue marbles, and 2 green marbles, without replacement, is (4/9) * (3/8) or approximately 0.1667.

5. In a game, you have a bag containing 10 yellow balls, 8 green balls, and 7 blue balls. If you draw two balls without replacement, what is the probability of drawing two green balls?

In [6]:
# set ball draw options
ball_draw = ["yellow","green", "blue"]
weights = [10/25, 8/25, 7/25]
n_balls = 2

# run iterations and make random choices based on given weights
rand_selection = np.random.choice(a= ball_draw, size=(iterations, n_balls), p=weights)
rand_selection
                                
# count all green that appears twice
rand_selection = (rand_selection == "green").sum(axis=1) == 2

# get probability
rand_selection.mean()

0.10239

    - The probability of drawing two green balls from a bag containing 10 yellow balls, 8 green balls, and 7 blue balls, without replacement, is (8/25) * (7/24) or approximately 0.10239.

6. A spinner is divided into 8 equal sections numbered from 1 to 8. If you spin the spinner once, what is the probability of landing on an even number?

In [17]:
# spinner options
spinner = np.arange(1,9)

# run iterations and make random choices
rand_selection = np.random.choice(spinner, iterations)

# count evens in rand_selection
rand_selection = ((rand_selection % 2) == 0).sum()

# get probability
rand_selection.mean() / iterations

0.50268

    - The probability of landing on an even number when spinning a spinner divided into 8 equal sections numbered from 1 to 8 is 4/8 or 1/2 or approximately 0.50268.

7. A box contains 5 black socks and 3 white socks. If you randomly select two socks from the box without replacement, what is the probability of selecting two black socks?

In [37]:
# socks options
socks = ["black", "white"]
weights = [5/8, 3/8]

# run iterations and make random choices based on given weights
rand_selection = np.random.choice(socks, (iterations, n_socks), weights)

# count all socks where they are both back
rand_selection = (rand_selection == "black").all(axis=1).sum()

# get probability
rand_selection.mean() / iterations

0.25032

    - The probability of selecting two black socks from a box containing 5 black socks and 3 white socks, without replacement, is (5/8) * (4/7) or approximately 0.2509.

8. A deck of 52 playing cards is shuffled thoroughly and a card is drawn at random. What is the probability of drawing a face card (jack, queen, or king)?

In [13]:
# cards options
cards = ["face","other"]
weights = [3/13, 10/13]

# run iterations and make random choices based on given weights
rand_selection = np.random.choice(a=cards, size=iterations, p=weights)

# count face cards
rand_selection = (rand_selection == "face").sum()

# get probability
rand_selection.mean() / iterations

0.23265

    - The probability of drawing a face card (jack, queen, or king) from a shuffled deck of 52 playing cards is 12/52 or 3/13 or approximately 0.23265.

9. A bag contains 6 red balls, 4 blue balls, and 5 green balls. If you randomly select two balls with replacement, what is the probability of selecting a red ball on the first draw and a green ball on the second draw?

In [34]:
# set ball draw options
ball_draw = ["red","bue", "green"]
weights = [6/15, 4/15, 5/15]
n_balls = 2

# run iterations and make random choices based on given weights
rand_selection = np.random.choice(a= ball_draw, size=(iterations, n_balls), p=weights)
rand_selection
                                
# count all red that appears at the first index
# count all greens that appears at the second index
rand_selection_1 = (rand_selection == "red").sum(axis=0)
rand_selection_2 = (rand_selection == "green").sum(axis=0)

# get probability
red = rand_selection_1[0] / iterations
green = rand_selection_2[1] / iterations

# multipy the probalilities
red * green

0.134905924

    - The probability of selecting a red ball on the first draw and a green ball on the second draw from a bag containing 6 red balls, 4 blue balls, and 5 green balls, with replacement, is (6/15) * (5/15) or approximately 0.1349.

10. In a game, you have a deck of 30 cards, numbered from 1 to 30. If you draw a card at random, what is the probability of drawing a multiple of 3?

In [37]:
# deck options
deck = np.arange(1,31)

# run iterations and make random choices
rand_selection = np.random.choice(a=deck, size= iterations)

# count all numbers divisible by 3
rand_selection = ((rand_selection % 3) == 0).sum()

# get probability
rand_selection / iterations

0.33385

    - The probability of drawing a multiple of 3 from a deck of 30 cards, numbered from 1 to 30, is 10/30 or 1/3 or approximately 0.3338.